services/rfq/relayer/service/otel.go

```diff
@@ -0,0 +1,82 @@
+package service
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cornelk/hashmap"
+	"github.com/synapsecns/sanguine/core/metrics"
+	"github.com/synapsecns/sanguine/ethergo/signer/signer"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+const meterName = "github.com/synapsecns/sanguine/services/rfq/relayer/service"
+
+// generate an interface for otelRecorder that exports the public method.
+// this allows us to avoid using recordX externally anad makes the package less confusing.
+//
+// =============================================================================
+// =============================================================================
+// IMPORTANT: DO NOT REMOVE THIS COMMENT.
+// NOTICE: PLEASE MAKE SURE YOU UPDATE BOTH THE DOCS AND THE GRAFANA DASHBOARD (IF NEEDED) AFTER UPDATING METRICS.
+// =============================================================================
+// =============================================================================
+//
+//go:generate go run github.com/vburenin/ifacemaker -f otel.go -s otelRecorder -i iOtelRecorder -p service -o otel_generated.go -c "autogenerated file"
+type otelRecorder struct {
+	metrics metrics.Handler
+	// meter is the metrics meter.
+	meter metric.Meter
+	// statusCountGauge is the gauge for the status.
+	statusCountGauge metric.Int64ObservableGauge
+	// statusCounts is used for metrics.
+	// status -> count
+	statusCounts *hashmap.Map[int, int]
+	// signer is the signer for signing transactions.
+	signer signer.Signer
+}
+
+func newOtelRecorder(meterHandler metrics.Handler, signer signer.Signer) (_ iOtelRecorder, err error) {
+	or := otelRecorder{
+		metrics:      meterHandler,
+		meter:        meterHandler.Meter(meterName),
+		statusCounts: hashmap.New[int, int](),
+		signer:       signer,
+	}
+
+	or.statusCountGauge, err = or.meter.Int64ObservableGauge("status_count")
+	if err != nil {
+		return nil, fmt.Errorf("could not create status count gauge")
```

Comment:
error should be wrapped here (e.g. `fmt.Errorf("could not create status count gauge: %w")` but in cases where it's not it's idiomatic to use `errors.New()` instead of fmt.Errorf